Feature develop game

santiagorodriguezbermudez opened the pull request:
## Feature Develop Tic Tac Toe Game

### On this feature we worked on:
- [x] Implemented JS Player factory function.
- [x] Implemented JS Game Logic and Game Board Modules
- [x] Added styling for playing the game

santiagorodriguezbermudez committed:
Added game logic and first objects to script file

santiagorodriguezbermudez committed:
Added html first structure to project

Commit:
Update page structure

Commit:
Add newGame method to start the game

Commit:
Move the module and methods to the separate files

santiagorodriguezbermudez committed:
Fixed main module functions to start the game

santiagorodriguezbermudez committed:
Added game logic for the entire game

santiagorodriguezbermudez committed:
Style messages for users

santiagorodriguezbermudez committed:
Fixed html to gain access to module

santiagorodriguezbermudez committed:
Added delete method for restarting the board

Commit:
Add start screen structure and styles

Commit:
Add styles for the players card

Commit:
Add display and delete players methods

Commit:
Fix the container id name

Commit:
Add classes to the player card blocks and to the cells

Commit:
Set data attribute for the default view of the cells

Commit:
Update html structure of the game page

Commit:
Add calls for the players card

Commit:
Add minor fixes on js file styles

Commit:
Add toggle class method to render with different styles player tokens

Commit:
Add disable cell method and calls to disallow clicking to the cells when the game finished

Commit:
Add toggle start elements method to show logo and buttons during the game with different styles

Commit:
Add styles and fonts

Commit:
Remove commented line

Commit:
Update styles of the button and cells for the hover effects

santiagorodriguezbermudez committed:
Fixed the tie logic conditional

santiagorodriguezbermudez committed:
Change logic for styles for new game

santiagorodriguezbermudez committed:
Added github actions files

santiagorodriguezbermudez committed:
Fixed linters for the board js file

santiagorodriguezbermudez committed:
Fixed linters for game file

santiagorodriguezbermudez committed:
Fixed linters for main and player's files.

santiagorodriguezbermudez committed:
Fixed linters to scss file and css output

santiagorodriguezbermudez committed:
Updated Readme File with live demo link

stratospherique reviewed (changes requested):
### Great start @team :clap: Impressive implementation of the game :1st_place_medal: However, some changes need to be applied in order to get approval for this project. Check :arrow_down: 

##### Application (Required)
* [x] It has a function with the only responsibility of rendering the contents of the gameboard array to the webpage ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/tic_tac_toe/render_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/tic-tac-toe-javascript))
- Be aware of circular import. Check out this [article](https://spin.atomicobject.com/2018/06/25/circular-dependencies-javascript/#:~:text=js%20does%20support%20circular%20require,work%20correctly%20within%20an%20application.%E2%80%9D) 
* [x] It includes a button to reset the game ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/tic_tac_toe/ui_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/tic-tac-toe-javascript))
- It would be better if the user won't have to go through game initiation phase each time he wants to play (e.g setting names). Or at least, introduce a choice

#### Gameboard (required) :warning: 

* [x] It uses an array inside of a Gameboard object to store the 'moves' ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/tic_tac_toe/objects_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/tic-tac-toe-javascript))
- You can add a Gameboard module/factory, and make 'board.js' as the one responsible for controlling the flow of the game.
##### General (required)
* [x] The project has a descriptive README.md ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/descriptive_readme.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))
- Your game is amazing, you should include screenshots of it and a description section.
### That's it :relieved: When finished, kindly submit a code re-review request :pray: 

Happy coding :v: 
**_Ahmed Mahfoudh_**

santiagorodriguezbermudez committed:
Updated readme file with description and screenshot

Commit:
Refactor code for the requested changes v1

santiagorodriguezbermudez committed:
Added reset button for game

santiagorodriguezbermudez committed:
Reset method for new reset button

santiagorodriguezbermudez committed:
Decoupled board and game modules

santiagorodriguezbermudez committed:
Change import  comments on for linters

Commit:
Update styles

Commit:
Minor fixes

Commit:
Fix offences for the linters

Commit:
Chane the stylelint rule in accordance with the instructions https://github.com/microverseinc/linters-config/tree/master/html-css

santiagorodriguezbermudez committed:
Fixed style linters

santiagorodriguezbermudez committed:
Change linter github action config

santiagorodriguezbermudez committed:
Change root path for live demo

santiagorodriguezbermudez committed:
Updated the screenshot with new reset button

santiagorodriguezbermudez commented:
## Code review # 1

### In this CR we worked on:
- [x] Change the game logic to just contain game logic
- [x] Change the game board to contain everything related to the DOM. Including the game array. 
- [x] Added a reset button to start the game over with the same players. 
- [x] Updated the ReadME file with the game screenshots.

wathigo reviewed (changes requested):
### Changes Required.
Great work on implementing the suggested changes

**REQUIRED**.
There is an error preventing me from starting the game: See :point_down: 
![rev](https://user-images.githubusercontent.com/44072711/92034747-5965f780-ed76-11ea-9770-0b5cc4c94462.gif)
Let's fix this bug.

**After working on the changes, submit another request for review.**

**Happy Hacking :computer: 
Simon Wathigo.**

santiagorodriguezbermudez commented:
### Code review # 2

The live demo and local repo is working for us. See this [video](https://www.loom.com/share/3c6e0d34c68841a2b2d1060885170207) for reference.

wathigo reviewed (approved):
### Approved :heavy_check_mark: 
Great job guys!
The game executes as expected.
You have also put a lot of effort into UX.

**You may move on.**

**Good Luck :four_leaf_clover: 
Simon Wathigo.**
